uuid.go

```diff
@@ -4,22 +4,36 @@ import (
 	"crypto/rand"
 	"encoding/hex"
 	"fmt"
+	"io"
 )
 
 // GenerateRandomBytes is used to generate random bytes of given size.
 func GenerateRandomBytes(size int) ([]byte, error) {
+	return generateRandomBytesByReader(size, rand.Reader)
+}
+
+func generateRandomBytesByReader(size int, reader io.Reader) ([]byte, error) {
 	buf := make([]byte, size)
-	if _, err := rand.Read(buf); err != nil {
+	if _, err := io.ReadFull(reader, buf); err != nil {
```

Comment:
it is kind of redundant to force crypto/rand's reader through ReadFull, logic could split here depending on whether or not reader is rand.Reader. if it is rand.Reader then use `reader.Read()` otherwise use `io.Readfull()`

Reply:
I think it's fine to have crypto/rand use ReadFull